pyccel/parser/semantic.py

```diff
@@ -853,8 +859,44 @@ def _visit_CommentBlock(self, expr, **settings):
         return expr
     def _visit_AnnotatedComment(self, expr, **settings):
         return expr
+
     def _visit_OmpAnnotatedComment(self, expr, **settings):
+        code = expr._user_nodes
+        code = code[-1]
+        index = code.body.index(expr)
+        combined_loop = expr.combined and ('for' in expr.combined or 'distribute' in expr.combined or 'taskloop' in expr.combined)
+
+        if isinstance(expr, (OMP_Sections_Construct, OMP_Single_Construct)) \
+           and expr.has_nowait:
+            for i, node in enumerate(code.body):
+                if i > index:
+                    if isinstance(node, Omp_End_Clause):
+                        if node.txt.startswith(expr.name, 4):
+                            node.has_nowait = True
+
+        if isinstance(expr, (OMP_For_Loop, OMP_Simd_Construct,
+                            OMP_Distribute_Construct, OMP_TaskLoop_Construct)) or combined_loop:
+            msg = "Statement after {} must be a for loop.".format(type(expr).__name__)
+            if index == (len(code.body) - 1):
+                errors.report(msg, symbol=type(expr).__name__,
+                severity='fatal', blocker=self.blocking)
+            for i, node in enumerate(code.body):
+                if i == index + 1:
```

Comment:
Why is the for loop necessary here? Can't you just do:
```
if len(code.body) > index + 1:
    i = index + 1
```
?

Reply:
Ah sorry, I missed:
```
elif isinstance(node, (Comment, CommentBlock, Pass)):
                        index += 1
```

Reply:
But I still think this could be simplified with a `while` block or at least by only starting the for loop at index + 1